include/oneapi/tbb/mutex.h

```diff
@@ -36,9 +36,7 @@ class mutex {
     };
 
     //! Destructor
-    ~mutex() {
-        __TBB_ASSERT(!my_flag.load(std::memory_order_relaxed), "destruction of an acquired mutex");
-    }
+    ~mutex() = default;
```

Comment:
> the destructor of the mutex might be called before real mutex EOL

How come is that? Can you please elaborate on this?

Reply:
@aleksei-fedotov @pavelkumbrasev,
It seems like Aleksei is right - I have double checked and `mutex` is not trivial type and strictly speaking, accessing memory before the constructor and after the destructor is UB.
`mutex` is not trivial because its default ctor is not trivial. First of all, we have itt call inside of the body and secondly, we initialize `waitable_atomic<bool>` inside of the mutex with `0` which is not trivial.
`waitable_atomic<bool>` is also not a trivial type but it can be easily made trivial by removing `{}` from the atomic variable inside.
I have also checked standard traits `std::is_trivial_v<mutex>` and `std::is_trivial_v<waitable_atomic<bool>>` - both of them are `false`.

It seems like we need to add a TODO comment and resolve this issue separately because it is more complex then just making the destructor trivial. I think we also can return the assert and add a TODO comment because it is not related to this patch.